### ISSUE TITLE
getting ready for 1.0 CR

### ISSUE THREAD

**Issue by gavinking** [hibernate/hibernate-reactive#599]
So now that we finally have full coverage of the JPA mapping annotations, I think we should be just about ready for a 1.0 release.

Here's a list of the issues I think we should try to sort out before the release:

https://github.com/hibernate/hibernate-reactive/milestone/2

Note that:

- #565 depends on my patch to Hibernate core, and therefore on a new release of core.
- #570 depends on a release of Vert.x, but hope that drops soon. If it doesn't, no problem, it's apparently not a huge issue.
- #313 is an enhancement that I think would be nice to have, and doesn't look very hard.
- #441 impacts the API, so we need to make a decision.
- #592 is just testing, but it's always nice to know what works and what doesn't work.

Is there anything else that we should fix before doing a candidate release?

**Comment by gavinking** [comment]
Note that I've also created a second milestone for thing I think we can do after 1.0:

https://github.com/hibernate/hibernate-reactive/milestone/3

But if anyone thinks one of those things need doing *now*, well, they had better start work on it ;-)

**Comment by DavideD** [comment]
Looks great!
Thanks a lot

**Comment by gavinking** [comment]
AFAICT, all we're waiting on now is a release of Hibernate core. @Sanne WDYT?

**Comment by Sanne** [comment]
Ok, there's a couple more fixes (unrelated to HR) that need herding but yes let's try to trigger a core release tonight/tomorrow.

**Comment by gavinking** [comment]
Awesome.

**Comment by gavinking** [comment]
Alright, we have done everything necessary for a 1.0 CR release.

However, I'm most of the way through implementing #596, so if it keeps going well, I might just about be able to sneak that into the release.

Also I need to write an announcement.

**Comment by gavinking** [comment]
Oh and I guess I promised to reorg the `example` dir.

**Comment by gavinking** [comment]
Let's put this release on hold while we work on #642.

**Comment by DavideD** [comment]
Sounds good to me

**Comment by Sanne** [comment]
What about the Vert.x version?  Quarkus is about to upgrade to Vert.x 4, so we'll need to follow and that's not an appropariate change for the time between say CR1 and Final.

I was assuming we were going to wait and ship our first Final as based on Vert.x 4, but of course that could be a 1.1+  - assuming we're in agreement that it's an acceptable change for a minor.

**Comment by DavideD** [comment]
I'd like to wait for Vert.x 4, but is there an estimate about when it is going to be ready in Quarkus?

**Comment by Sanne** [comment]
The original estimate was "March 2021" but I'm sure it was optimistic; I'll get an update tomorrow 2pm

**Comment by gavinking** [comment]
Is there any reason we can't release 1.0 for Vert.x 3 and then 1.1 for Vert.x 4?

**Comment by Sanne** [comment]
> Is there any reason we can't release 1.0 for Vert.x 3 and then 1.1 for Vert.x 4?

Sure we can. That's one of the options I suggested above.

Just bear in mind, 1.1 will likely need to follow 1.0 very quickly. I have no problem with that, just making sure we're on the same page.

**Comment by gavinking** [comment]
> Just bear in mind, 1.1 will likely need to follow 1.0 very quickly. I

That's totally fine by me.

**Comment by Sanne** [comment]
> The original estimate was "March 2021" but I'm sure it was optimistic; I'll get an update tomorrow 2pm

Ok got the info: it's going to take a few weeks more at least. So let's not hold for that, and we'll follow up with a 1.1 whenever we need a vert.x 4 build.

**Comment by gavinking** [comment]
Guess I should start writing the announcement.

**Comment by gavinking** [comment]
OK, this is all done. We now just need to go ahead and release 1.0 CR1.
